l10n_it_fatturapa/migrations/12.0.3.1.0/pre-migration.py

```diff
@@ -0,0 +1,15 @@
+#  Copyright 2022 Simone Rubino - TAKOBI
+#  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+def migrate(cr, installed_version):
+    openupgrade.rename_columns(
+        cr,
+        {
+            'res_company': [
+                ('phone_electronic_invoice', None),
```

Comment:
A cosa serve questo?

Reply:
Questo salva i valori della colonna `phone_electronic_invoice` in un'altra colonna _legacy_ di `res_company`, il cui nome è deciso da `openupgrade.get_legacy_name`.
Dopo l'aggiornamento del modulo (`post-migration`) i valori di questa colonna _legacy_ vengono portati sul partner collegato all'azienda con `openupgrade.move_field_m2o`.